### PR TITLE
[Stable fix] Type propagation respecting type hierarchy

### PR DESCRIPTION
**Pull request by kasper-piskorski**

### CODE REVIEW

**Comment by kasper-piskorski** [comment]
please retest this please - ai.grakn.test.engine.controller.TasksControllerTest.afterSendingTaskWithRunAt_ItIsDelayedInStorage

File: grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java [hunk 1]
```diff
@@ -345,7 +348,7 @@ public void inferringSpecificResourceValue() {
     }
 
     @Test //Expected result: Two answers obtained only if the rule query containing sub type is correctly executed.
-    public void instanceTypeHierarchyRespected(){
+    public void instanceTypeHierarchyRespected_queryHasSuperTypes(){
```
Comment (fppt) [inline grakn-test/src/test/java/ai/grakn/test/graql/reasoner/ReasoningTests.java]: 👍

**Comment by fppt** [comment]
@kasper-piskorski This fix needs to go into master now. Please create a merge PR into master and reference it here.

**Comment by fppt** [comment]
Merged into master here: https://github.com/graknlabs/grakn/pull/1663

**Pull request merged**